File: cpp/Memory.cpp
#include <stdio.h>
using namespace std;

#include "Hardware.cpp"

class Memory : public Hardware
{
private:
    int frequency;
    int memorySize;
    bool supportCuda;

public:
    Memory() {}
    Memory(int frequency, int memorySize, bool supportCuda)
    {
        this->frequency = frequency;
        this->memorySize = memorySize;
        this->supportCuda = supportCuda;
    }

    int getFrequency()
    {
        return this->frequency;
    }
    void setFrequency(int frequency)
    {
        this->frequency = frequency;
    }

    int getMemorySize()
    {
        return this->memorySize;
    }
    void setMemorySize(int memorySize)
    {
        this->memorySize = memorySize;
    }

    bool isSupportCuda()
    {
        return this->supportCuda;
    }
    void setSupportCuda(bool supportCuda)
    {
        this->supportCuda = supportCuda;
    }

    void printMemory()
    {
        cout << "Frequency: " << getFrequency() << " Mhz" << endl;
        cout << "Memory Size: " << getMemorySize() << " GB" << endl;
        cout << "Support Cuda: ";
        if (isSupportCuda())
        {
            cout << "Yes" << endl;
        }
        else
        {
            cout << "No" << endl;
        }
    }
    ~Memory() {}
};

File: python/Memory.py
from pickle import FALSE
from Hardware import Hardware

class Memory(Hardware):
    def __init__(self):
        self.frequency = 0
        self.memorySize = 0
        self.supportCuda = False

    def getFrequency(self):
        return self.frequency

    def setFrequency(self, frequency):
        self.frequency = frequency

    def getMemorySize(self):
        return self.memorySize

    def setMemorySize(self, memorySize):
        self.memorySize = memorySize

    def isSupportCuda(self):
        return self.supportCuda

    def setSupportCuda(self, supportCuda):
        self.supportCuda = supportCuda

    def printMemory(self):
        print("Frequency : {}Mhz".format(self.getFrequency()))
        print("Memory Size : {}GB".format(self.getMemorySize()))
        print("Support Cuda: ", end="")
        if self.isSupportCuda():
            print("Yes")
        else:
            print("No")